src/Microsoft.AspNet.Hosting/Startup/StartupLoader.cs

```diff
@@ -94,9 +90,9 @@ private object Invoke(MethodInfo methodInfo, object instance, IApplicationBuilde
             string environmentName,
             IList<string> diagnosticMessages)
         {
-            if (String.IsNullOrEmpty(applicationName))
+            if (string.IsNullOrEmpty(applicationName))
             {
-                return _next.LoadStartup(applicationName, environmentName, diagnosticMessages);
+                throw new ArgumentException("applicationName");
```

Comment:
Wrong parameter to this exception. ArgException takes either "message" or "message, paramName". Use this message: https://github.com/aspnet/Mvc/blob/f6e701b2b2d01ae068b144416aeac080c623bc17/src/Microsoft.AspNet.Mvc.Core/Resources.resx#L205-L207